src/audio/kpb.c

```diff
@@ -512,7 +511,7 @@ static int kpb_copy(struct comp_dev *dev)
 
 	/* Sink and source are both ready and have space */
 	copy_bytes = MIN(sink->free, source->avail);
-	assert(!memcpy_s(sink->w_ptr, copy_bytes, source->r_ptr, copy_bytes));
+	memcpy(sink->w_ptr, source->r_ptr, copy_bytes);
```

Comment:
It is highly desirable to use buffer_write_frag API here to handle potential buffer wrap-ups.

Reply:
Yes I agree, added function which do care about it. Once this PR is merged I will create some container which keeps different copy function for different format and they will be selected by looking up `cd->config.sampling_width`